R/LSTconfidenceIntervals.R

```diff
@@ -0,0 +1,407 @@
+  } else {
+    loopingVector <- 1:options$dataPlotShowN
+  }
+  
+  listCounter <- 1
+  for (i in loopingVector) {
+    
+    if (addDots && listCounter == 2) {
+      plotList[[listCounter]] <- ggplot2::ggplot(data = data.frame(x = c(-1,1), y = c(-1,1))) + ggplot2::theme_void() +
+        ggplot2::annotate("text", x = -3, y = -2.5, 
+                          label = gettextf("..."), 
+                          size = 60)
+    } else {
+      
+      thisRainData <- data.frame(y = unlist(jaspContainer[["simulatedDatasets"]][["object"]][[i]]), 
+                                 group = rep(1, options$n))  
+      
+      plotList[[listCounter]] <- try(jaspTTests::.descriptivesPlotsRainCloudFill(thisRainData, "y", "group", 
+                                                                                 yLabel = "Dependent", 
+                                                                                 xLabel = paste("Repetition", i),
+                                                                                 testValue = options$mu,
+                                                                                 addLines = FALSE, horiz = FALSE))
+      if(isTryError(plotList[[listCounter]]))
+        matrixPlot$setError(.extractErrorMessage(plotList[[listCounter]]))
+      
+    }
+    listCounter <- listCounter + 1
+  }
+  
+  p <- jaspGraphs::ggMatrixPlot(matrix(plotList, nrow = nRows, ncol = nCols, byrow = TRUE))
+  matrixPlot$plotObject <- p
+  jaspContainer[["containerRainCloudPlots"]][["plotObject"]] <- matrixPlot
+  
+  return()
+}
+
+.plotConvergencePlot <- function(jaspContainer, options) {
+  if (!is.null(jaspContainer[["containerConvergencePlot"]]) || !options$convergencePlot)
+    return()
+  
+  containerConvergencePlot <- createJaspContainer(title = gettext("Convergence Plot"))
+  jaspContainer[["containerConvergencePlot"]] <- containerConvergencePlot
+  containerConvergencePlot$dependOn(c("convergencePlot", "convergenceZoomIn", "convergenceZoomMargin"))
+  containerConvergencePlot[["convergencePlot"]] <- createJaspPlot(title = "", width = 480, height = 320) 
+  
+  meanCI <- jaspContainer[["computedConfidenceIntervals"]][["object"]]
+  
+  myTicks <- jaspGraphs::getPrettyAxisBreaks(1:options$nReps)
+  myTicks[1] <- 1
+
+  if (options$convergenceZoomIn) {
+    myYlim <- c(max(0, options$confidenceIntervalInterval - options$convergenceZoomMargin),
+                min(1, options$confidenceIntervalInterval + options$convergenceZoomMargin))  
+  } else {
+    myYlim <- c(0, 1)
+  }
+  
+  cumSuccessProportion <- cumsum(meanCI$successfulCI)/1:options$nReps
+  simulMatrix <- matrix(0, nrow = 1000, ncol = options$nReps)
+  
+  for (i in 1:options$nReps){
+    simulMatrix[ , i] <- rbeta(1e3, cumSuccessProportion[i]*i+1, i-cumSuccessProportion[i]*i+1)
+  }
+  credInt <- apply(simulMatrix, 2, HDInterval::hdi) # record the credibility interval
+  y.upper <- credInt[1,]
+  y.lower <- credInt[2,]
```

Comment:
```suggestion
  yUpper <- credInt[1,]
  yLower <- credInt[2,]
```
for consistency :slightly_smiling_face: